src/main/java/org/jabref/gui/texparser/ParseTexDialogViewModel.java

```diff
@@ -117,18 +118,31 @@ public void searchButtonClicked() {
                           noFilesFound.set(true);
                           searchInProgress.set(true);
                           successfulSearch.set(false);
-
                       })
                       .onFinished(() -> searchInProgress.set(false))
                       .onSuccess(newRoot -> {
                           root.set(newRoot);
                           noFilesFound.set(false);
                           successfulSearch.set(true);
                       })
-                      .onFailure(dialogService::showErrorDialogAndWait)
+                      .onFailure(this::handleFailure)
                       .executeWith(taskExecutor);
     }
 
+    private void handleFailure(Exception exception) {
+        root.set(null);
+        noFilesFound.set(true);
+        searchInProgress.set(false);
+        successfulSearch.set(false);
```

Comment:
`root` is already `null` from `onRunning()` event. Same for `noFilesFound.set(true)` and `successfulSearch.set(false)`. Moreover, `searchInProgress.set(false)` is already done in the `onFinished()` event, which is executed whether the task is finished with an error or if it succeeds.

I think that these four lines could be removed.

Reply:
Thanks! fixed in f6790ca2192e9a22711a91c8054bb352af02702a